apollo4x: build failure on `kernel/cache/` Twister tests

**Describe the bug**

The `kernel/cache` test suite fails on the `apollo4p_evb` and `apollo4p_blue_kxr_evb` boards due to missing symbols. It appears that (the HAL?) does not declare the `SCB_`* cache control functions that are used by `arch/arm/core/cortex_m/cache.c` when `CONFIG_CACHE_MANAGEMENT` is defined. 

**To Reproduce**

Run `west twister -c -T tests/kernel/cache/ -p apollo4p_evb`. 
On current Zephyr main (f5ce0476c5d, https://github.com/zephyrproject-rtos/hal_ambiq/commit/f6858cb3fe3d44945a23d114b7feeeac22ce71cf), tests fail to build with this error:

```
FAILED: zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/cache.c.obj 
ccache zephyr-sdk-0.16.4/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DKERNEL -DK_HEAP_MEM_POOL_SIZE=0 -DTC_RUNID=c0f20f410652dfe4df7d454c06f392ed -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -Izephyr/kernel/include -Izephyr/arch/arm/include -Izephyr/include -Izephyr/twister-out/apollo4p_evb/tests/kernel/cache/kernel.cache.api.minimallibc/zephyr/include/generated -Izephyr/soc/ambiq -Izephyr/soc/ambiq/apollo4x/. -Izephyr/soc/ambiq/. -Izephyr/subsys/testsuite/include -Izephyr/subsys/testsuite/ztest/include -Icore/include -Imodules/hal/cmsis/CMSIS/Core/Include -Izephyr/modules/cmsis/. -Imodules/hal/ambiq/CMSIS/AmbiqMicro/Include -Imodules/hal/ambiq/utils -Imodules/hal/ambiq/mcu/apollo4p -Imodules/hal/ambiq/mcu/apollo4p/hal -Imodules/hal/ambiq/components/bluetooth -isystem zephyr/lib/libc/minimal/include -isystem zephyr/lib/libc/common/include -isystem zephyr-sdk-0.16.4/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/include -isystem zephyr-sdk-0.16.4/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/include-fixed -Wshadow -fno-strict-aliasing -Werror -Os -imacros zephyr/twister-out/apollo4p_evb/tests/kernel/cache/kernel.cache.api.minimallibc/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m4 -mthumb -mabi=aapcs -mfp16-format=ieee --sysroot=zephyr-sdk-0.16.4/arm-zephyr-eabi/arm-zephyr-eabi -imacros zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=zephyr/tests/kernel/cache=CMAKE_SOURCE_DIR -fmacro-prefix-map=zephyr=ZEPHYR_BASE -fmacro-prefix-map=ZephyrCore=WEST_TOPDIR -ffunction-sections -fdata-sections -std=c99 -nostdinc -MD -MT zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/cache.c.obj -MF zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/cache.c.obj.d -o zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/cache.c.obj -c zephyr/arch/arm/core/cortex_m/cache.c
zephyr/arch/arm/core/cortex_m/cache.c: In function 'arch_dcache_enable':
zephyr/arch/arm/core/cortex_m/cache.c:20:9: error: implicit declaration of function 'SCB_EnableDCache' [-Werror=implicit-function-declaration]
   20 |         SCB_EnableDCache();
      |         ^~~~~~~~~~~~~~~~
```
(and all other `SCB_`* functions).

**Impact**

CI failures ([sample log](https://github.com/zephyrproject-rtos/zephyr/actions/runs/9079517508/job/24949484379?pr=72358)) on recent LLEXT PRs (such as #72358).

--- a/boards/ambiq/apollo4p_blue_kxr_evb/apollo4p_blue_kxr_evb.yaml
+++ b/boards/ambiq/apollo4p_blue_kxr_evb/apollo4p_blue_kxr_evb.yaml
@@ -20,7 +20,3 @@ testing:
   ignore_tags:
     - net
 vendor: ambiq
-# Provisional hack to prevent tests being run in this board, as it fails in many test & samples:
-twister: false
-# Once https://github.com/zephyrproject-rtos/zephyr/issues/74212, 73443 & 72775 are fixed
-# this should be removed

--- a/boards/ambiq/apollo4p_evb/apollo4p_evb.dts
+++ b/boards/ambiq/apollo4p_evb/apollo4p_evb.dts
@@ -14,6 +14,7 @@
 		zephyr,console = &uart0;
 		zephyr,shell-uart = &uart0;
 		zephyr,uart-pipe = &uart0;
+		zephyr,flash-controller = &flash;
 	};
 
 	aliases {
@@ -101,6 +102,23 @@
 	pinctrl-0 = <&mspi2_default>;
 	pinctrl-names = "default";
 	status = "okay";
+};
+
+&flash0 {
+	erase-block-size = <2048>;
+	write-block-size = <16>;
+
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		/* Set 16KB of storage at the end of the 1952KB of flash */
+		storage_partition: partition@1e4000 {
+			label = "storage";
+			reg = <0x001e4000 0x4000>;
+		};
+	};
 };
 
 &gpio0_31 {

--- a/boards/ambiq/apollo4p_evb/apollo4p_evb.yaml
+++ b/boards/ambiq/apollo4p_evb/apollo4p_evb.yaml
@@ -20,7 +20,3 @@ testing:
     - net
     - bluetooth
 vendor: ambiq
-# Provisional hack to prevent tests being run in this board, as it fails in many test & samples:
-twister: false
-# Once https://github.com/zephyrproject-rtos/zephyr/issues/74212, 73443 & 72775 are fixed
-# this should be removed

--- a/soc/ambiq/apollo4x/Kconfig
+++ b/soc/ambiq/apollo4x/Kconfig
@@ -11,6 +11,4 @@ config SOC_SERIES_APOLLO4X
 	select CPU_HAS_ARM_MPU
 	select HAS_SWO
 	select AMBIQ_HAL
-	select CPU_HAS_DCACHE
-	select CPU_HAS_ICACHE
 	select HAS_PM

--- a/west.yml
+++ b/west.yml
@@ -147,7 +147,7 @@ manifest:
       groups:
         - hal
     - name: hal_ambiq
-      revision: 367ae6a0396e3074bebbd55ef72f8e577168e567
+      revision: e25327f026df1ee08f1bf01a4bbfeb5e5f4026f1
       path: modules/hal/ambiq
       groups:
         - hal

